src/browser-lib/single-spa-angular.ts

```diff
@@ -92,11 +91,33 @@ function mount(opts, props) {
         if (!module || typeof module.destroy !== 'function') {
           throw Error(`single-spa-angular: the opts.bootstrapFunction returned a promise that did not resolve with a valid Angular module. Did you call platformBrowser().bootstrapModuleFactory() correctly?`)
         }
-        opts.bootstrappedNgZone = module.injector.get(opts.NgZone)
+
+        const singleSpaPlatformLocation: SingleSpaPlatformLocation | null = module.injector.get(
+          SingleSpaPlatformLocation,
+          null,
+        );
+
+        // The user has to provide `BrowserPlatformLocation` only if his application uses routing.
+        // So if he provided `Router` but didn't provide `BrowserPlatformLocation` then we have to inform him.
+        if (opts.Router && singleSpaPlatformLocation === null) {
+          throw new Error(`	
+            single-spa-angular: could not retrieve extra providers from the platform injector. Did you call getSingleSpaExtraProviders() when creating platform?	
+          `);
+        }
+
+        const ngZone = module.injector.get(opts.NgZone);
+
+        if (singleSpaPlatformLocation !== null) {
+          singleSpaPlatformLocation.setNgZone(ngZone);
+          // Cleanup resources, especially remove event listeners thus they will not be added
+          // twice when application gets bootstrapped the second time.
+          module.onDestroy(() => singleSpaPlatformLocation.destroy());
+        }
+
+        opts.bootstrappedNgZone = ngZone;
         opts.bootstrappedNgZone._inner._properties[opts.zoneIdentifier] = true;
         window.addEventListener('single-spa:routing-event', opts.routingEventListener)
         opts.bootstrappedModule = module;
-        patchRouter(opts);
```

Comment:
Removing this is technically a breaking change. Before this PR, the back button bug was fixed without having to use the single-spa extra providers. But now we require users to use the extra single-spa providers.

Since it's a breaking change, it should cause a new major version of single-spa-angular@3. However, we're sort of boxed in with our versioning because single-spa-angular@4 already exists. So I think best option is to publish this as a 3.x minor and then try to document this as much as possible in release notes?